WebView/app/src/androidTest/java/com/android/samples/webviewdemo/MainActivityTest.kt

```diff
@@ -0,0 +1,159 @@
+      // to automate WebViews is through javascript, it must be enabled.
+        onWebView().forceJavascriptEnabled()
+    }
+
+    // Test to check that the drop down menu behaves as expected
+    @Test
+    fun dropDownMenu_SanFran() {
+        mainActivityRule.getActivity()
+        onWebView()
+            .withElement(findElement(Locator.ID, "location"))
+            .perform(webClick()) // Similar to perform(click())
+            .withElement(findElement(Locator.ID, "SF"))
+            .perform(webClick()) // Similar to perform(click())
+            .withElement(findElement(Locator.ID, "title"))
+            .check(webMatches(getText(), containsString("San Francisco")))
+    }
+
+    // Test for checking createJsObject
+    @Test
+    fun jsObjectIsInjectedAndContainsPostMessage() {
+    mainActivityRule.getActivity()
+    onWebView()
+        .check(
+            webMatches(
+                script("return jsObject && jsObject.postMessage != null;", castOrDie(Boolean::class.javaObjectType)),
+                `is`(true)
+            )
+        )
+    }
+
+    @Test
+    fun valueInCallback_compareValueInput_returnsTrue(){
+        mainActivityRule.getActivity()
+
+        // Setup
+        val webView = WebView(context)
+        val jsObjName = "jsObject"
+        val allowedOriginRules = setOf<String>("https://example.com")
+
+        // Create HTML
+        val htmlPage = "<!DOCTYPE html><html><body>" + "    <script>" + "        myObject.postMessage('hello');" + "    </script>" + "</body></html>"
+
+        // Create JsObject
+        MainActivity.createJsObject(
+            webView,
+            jsObjName,
+            allowedOriginRules
+        ) { message -> MainActivity.invokeShareIntent(message) }
+
+        //Inject JsObject into Html
+        webView.loadData(htmlPage, "text/html", "UTF-8")
+
+
+        //Call js code to invoke callback (in script tag of htmlPage)
+
+        // evaluate what comes out -> it should be hello
+        // *Note: "response from callback" is a place holder here I am unsure what should be placed there
+        assertEquals("response from callback", "hello")
+
+    }
+
+    @Test
+    // Checks that postMessage runs on the UI thread
+    fun checkingThreadCallbackRunsOn() {
+        mainActivityRule.getActivity()
+
+        // Setup
+        val webView = WebView(context)
+        val jsObjName = "jsObject"
+        val allowedOriginRules = setOf<String>("https://example.com")
+
+        // Create HTML
+        val htmlPage =
+            "<!DOCTYPE html><html><body>" + "    <script>" + "        jsObject.postMessage('hello');" + "    </script>" + "</body></html>"
+
+
+        // Create JsObject
+        MainActivity.createJsObject(
+            webView,
+            jsObjName,
+            allowedOriginRules
+        ) { message -> MainActivity.invokeShareIntent(message) }
+
+
+        // Inject JsObject into Html by loading it in the webview
+        webView.loadData(htmlPage, "text/html", "UTF-8")
```

Comment:
You should swap this out for `loadDataWithBaseUrl` and use `https://example.com"` as the baseUrl. Otherwise, the `allowedOriginRules` may prevent the object from being injected.